### PR TITLE
Fix using vault encrypted data with jinja2_native

### LINKED ISSUE #48950: jinja2 native with vault encrypted var
##### SUMMARY
Vault encrypted strings are not getting cast to string before passing as a parmater to a module when `jinja2_native = True`.

On the surface this looks related to #24425, however that predates `jinja2_native` and this bug is only visible when `jinja2_native = True`.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ansible-vault
ansible/lib/ansible/executor/module_common.py
jinja2_native strings
???

##### ANSIBLE VERSION

```paste below
ansible 2.7.1
  config file = /home/cognifloyd/p/theatro-ansiblehost/ansible-playbooks.git/test/vault-test-case/ansible.cfg
  configured module search path = [u'/home/cognifloyd/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/cognifloyd/v/st2/lib/python2.7/site-packages/ansible
  executable location = /home/cognifloyd/v/st2/bin/ansible
  python version = 2.7.15 (default, Sep 30 2018, 15:08:44) [GCC 6.4.0]
```

##### CONFIGURATION
```
DEFAULT_HOST_LIST(/home/cognifloyd/p/theatro-ansiblehost/ansible-playbooks.git/test/vault-test-case/ansible.cfg) = [u'/home/cognifloyd/p/theatro-ansiblehost/ansible-playbooks.git/test/vault-test-case/inventory.yaml']
DEFAULT_JINJA2_NATIVE(/home/cognifloyd/p/theatro-ansiblehost/ansible-playbooks.git/test/vault-test-case/ansible.cfg) = True
DEFAULT_VAULT_IDENTITY_LIST(/home/cognifloyd/p/theatro-ansiblehost/ansible-playbooks.git/test/vault-test-case/ansible.cfg) = [u'cognifloyd@prompt']
HOST_KEY_CHECKING(/home/cognifloyd/p/theatro-ansiblehost/ansible-playbooks.git/test/vault-test-case/ansible.cfg) = False
INVENTORY_ENABLED(/home/cognifloyd/p/theatro-ansiblehost/ansible-playbooks.git/test/vault-test-case/ansible.cfg) = [u'host_list', u'yaml']
```
(see the sample ansible.cfg files in the gist)

##### OS / ENVIRONMENT
I'm on a gentoo linux host, with ansible installed via pip in a virtualenv.

##### STEPS TO REPRODUCE
https://gist.github.com/cognifloyd/1a43f28c02e05556703c1c87035fac32

files in gist:
- `jinja2native_with_vaulted.playbook.yaml` (minimal test case)
- `inventory.yaml (the minimal inventory I used to isolate the test case)
- `*_ansible.cfg` (2 versions one with `jinja2_native = True`, and one with `jinja2_native = False`)
- `*_output` (2 copies, one for each ansible.cfg)

##### EXPECTED RESULTS
The playbook should succeed no matter if `jinja2_native` is enabled or not.
See `no_native_passes_output` in the gist for output when `jinja2_native = False`.
The same thing should happen when `jinja2_native = True`.

##### ACTUAL RESULTS
See `with_native_fails_output` in the gist for output when `jinja2_native = True`.

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY

Fixes #48950

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

lib/ansible/executor/module_common.py
lib/ansible/template/__init__.py

##### ADDITIONAL INFORMATION

### CODE REVIEW

**Comment by abadger** [comment]
Two comments.

* The easier comment is that we already have a custom json encoder.  We shouldn't have differences when we can help it.  If the other one needs to work on the remote systems and ansible vault code isn't available there, we should at least have the vault aware one inept from that one rather than from the vanilla base class.
* The second one is harder... This feels like it is being done in the wrong place.  It feels like the values should be unvaulted much earlier than this.   For instance, an action plugin won't necessarily use the code in module_common but could also need the unvaulted value to perform its work. So the parameters should be unvaulted before we get to this point.

**Comment by mkrizek** [comment]
> * The second one is harder... This feels like it is being done in the wrong place. It feels like the values should be unvaulted much earlier than this. For instance, an action plugin won't necessarily use the code in module_common but could also need the unvaulted value to perform its work. So the parameters should be unvaulted before we get to this point.

Okay, so it sounds like it should be unvaulted in templar, as is the case with "non-native"/default.

File: lib/ansible/template/native_helpers.py [hunk 1]
```diff
@@ -30,8 +32,13 @@ def ansible_native_concat(nodes):
 
     if len(head) == 1:
         out = head[0]
+
+        # TODO send unvaulted data to literal_eval?
```
Comment (cognifloyd) [inline lib/ansible/template/native_helpers.py:36]: I vault dictionaries and lists. With this allow for that?

**Review by cognifloyd — COMMENTED** [review]
Just wondering about vaulted data that is not a string.

**Pull request merged**